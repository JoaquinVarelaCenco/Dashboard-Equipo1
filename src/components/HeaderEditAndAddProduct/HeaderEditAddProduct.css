/* HEADER EDIT PRODUCTS */
.headerEditProduct{
    display: flex;
    align-self: center;
}

.headerEditProduct__btnDeleteContainer{
    align-self: center;
    width: 100%;
    text-align: end;
}

.headerEditProduct__btnDelete{
    height: 32px;
    width: 100px;
    padding: 8px;
    
    border: 1px solid rgba(0, 0, 0, 0.5);
    border-radius: 16px;
    margin-left: 10px;
    
}

.headerEditProduct h2{
    font-weight: 500;
}