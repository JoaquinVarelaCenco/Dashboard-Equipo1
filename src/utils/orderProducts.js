
// orderByKey - Ordena la lista de productos por una key
// recibe como parametros la lista y el string de la key por cual ordenar
export const orderByKey = (productList, key) =>{
    const orderer = productList.sort((a,b) =>{
        if(a[key] < b[key]){
            return 1;
        }else  if(a[key] > b[key]){
            return -1;
        }else{
            return 0;
        }
    })
    return orderer;
}

// orderByKey - Ordena la lista de productos por una key
// recibe como parametros la lista y el string de las keys, la segunda es por la cual se ordena
export const orderByKeyDouble = (productList, key1, key2) =>{
    const orderer = productList.sort((a,b) =>{
        if(a[key1][key2] < b[key1][key2]){
            return 1;
        }else  if(a[key1][key2] > b[key1][key2]){
            return -1;
        }else{
            return 0;
        }
    })
    return orderer;
}