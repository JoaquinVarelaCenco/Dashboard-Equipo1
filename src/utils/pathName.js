export const pathName = (page) =>{
    //Lógica para establecer el pathname en que nos encontramos
    let pageTitle, titleLink;

    if (page === "/products") {
        pageTitle = "Productos";
        titleLink =page;
    } else if (page.includes("/stores")) {
        pageTitle = "Tiendas";
        titleLink =page;
    } else if (page === "/products/new") {
        pageTitle = "Productos";
        titleLink ="/products";
    } else if (page.includes("/products/")) {
        pageTitle = "Productos";
        titleLink ="/products";
    }else{
        pageTitle = "¡Hola Olivia!";
        titleLink = "/home";
    }
    return [pageTitle, titleLink];

}
