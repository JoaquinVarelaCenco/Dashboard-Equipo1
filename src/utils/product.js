import { deleteProduct } from "../services/apiServices";


export function deleteProductFunction(id) {
    deleteProduct(id).then((res) => {
    });
}

export const sortByCount = (products) =>{
    const sorted = products.sort((a,b) =>{
    if(a.rating.count < b.rating.count){
        return 1;
    }else  if(a.rating.count > b.rating.count){
        return -1;
    }else{
        return 0;
    }
    });
    return sorted;
}

export const sortByStock = (products) =>{
    const sorted = products.sort((a,b) =>{
    if(a.stock < b.stock){
        return 1;
    }else  if(a.stock > b.stock){
        return -1;
    }else{
        return 0;
    }
    });
    return sorted;
}