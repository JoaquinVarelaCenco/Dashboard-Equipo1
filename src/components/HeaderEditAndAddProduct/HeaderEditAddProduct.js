import { Link } from 'react-router-dom';
import leftArrow from '../../assets/images/chevron-right (1).svg';
import { deleteProductFunction } from '../../utils/product';
import Button from '../Button/Button';
import './HeaderEditAddProduct.css'

const HeaderEditAddProduct = ({productContent, editProduct}) => {
  return (
    <>
          <div className="headerEditProduct">
            <img src={leftArrow} alt="" />
            <h2>{productContent}</h2>
          </div>
          {editProduct?
            <div className="headerEditProduct__btnDeleteContainer">
              <Link to={"/products"}>
                <Button 
                  classN={"headerEditProduct__btnDelete"}
                  title={"ELIMINAR"}
                  click={()=>deleteProductFunction(productContent)}
                />
              </Link>
          </div>
          :
          ""  
        }
    </>
  )
}


export default HeaderEditAddProduct