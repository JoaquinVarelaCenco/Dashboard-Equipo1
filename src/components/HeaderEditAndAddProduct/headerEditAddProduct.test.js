import { cleanup, render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event";
import { MemoryRouter } from "react-router-dom";
import HeaderEditAddProduct from "./HeaderEditAddProduct"
import { deleteProductFunction } from '../../utils/product';


let component;

jest.mock("../../utils/product")

describe('Test boton eliminar producto del componente Header', ()=>{

  beforeEach(()=>{

    component = render(
      <MemoryRouter>
        <HeaderEditAddProduct editProduct={true}>
        </HeaderEditAddProduct>
        </MemoryRouter>)
  })

  it('El botón Eliminar debe estar en el Header cuando se esta editando un producto', ()=>{

    expect(screen.getByText(/eliminar/i)).toBeInTheDocument;

  })

  it('El botón Eliminar debe ejecutar la funcion al hacerse click', ()=>{
    let btnEliminar = screen.getByRole('button');
    userEvent.click(btnEliminar);
    expect(deleteProductFunction).toHaveBeenCalled();
  })


  it('El botón Eliminar no debe estar en el Header cuando no se esta editando un producto', ()=>{
    cleanup();

    render(
        <MemoryRouter>
          <HeaderEditAddProduct editProduct={false}>
          </HeaderEditAddProduct>
          </MemoryRouter>)

    expect(screen.queryByText(/eliminar/i)).not.toBeInTheDocument;

  })
})