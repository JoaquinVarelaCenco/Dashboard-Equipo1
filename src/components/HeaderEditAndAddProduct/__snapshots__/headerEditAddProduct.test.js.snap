// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test formulario de productos Se debe renderiza correctamente 1`] = `
<div>
  <div
    class="headerEditProduct"
  >
    <img
      alt=""
      src="chevron-right.svg"
    />
    <h2 />
  </div>
  <div
    class="headerEditProduct__btnDeleteContainer"
  >
    <a
      href="/products"
    >
      <button
        class="headerEditProduct__btnDelete"
      >
        ELIMINAR
      </button>
    </a>
  </div>
</div>
`;
